### ISSUE TITLE
Document deployment patterns

### ISSUE THREAD

**Issue by onbjerg** [onbjerg/foundry-book#183]
From @gakonst:

>Deployment patterns a-la https://github.com/gakonst/v3-periphery/blob/main/contracts/foundry-tests/utils/Deploy.sol

**Comment by Perelyn-sama** [comment]
Wouldn't forge script and forge create fit this ?

**Comment by gakonst** [comment]
Yeah, this is now something we'd want as a tutorial for `forge script`, e.g. https://github.com/joshieDo/v3-core/blob/main/deploy/deploy.sol

**Comment by Perelyn-sama** [comment]
What should be the level of the smart contract in the tutorial, does it have to be as complex as the uniswap v3 contracts?

**Comment by gakonst** [comment]
Up to you!

Ideally we have more than one example which starts simple and incrementally gets more complex.

**Comment by Perelyn-sama** [comment]
Submitting a pr in 48hrs
